src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/data-broker-profiles/welcome-to-premium/page.tsx

```diff
@@ -25,6 +26,11 @@ export default async function WelcomeToPremium() {
     redirect("/redesign/user/dashboard/");
   }
 
+  // The user may have subscribed and just need their session updated - they will be redirected back to try again if it looks valid.
+  if (!hasPremium(session.user)) {
+    redirect(`${process.env.NEXTAUTH_URL}/redesign/user/dashboard/subscribed`);
+  }
```

Comment:
Wouldn't we be passing this URL as a `redirect_uri` to subplat or FxA, instead of redirecting the user there ourselves?

Reply:
I think this would make more sense, I already provided them the above URL before I realized we couldn't easily update from the server side, but we can have it changed.

Reply:
OK one concern I had was if users subscribed first, but I don't think that actually matters, we can just have this "subscribed" endpoint be the place to start for subscription users.

I'm open to changing the URL FWIW, I only chose this one for convenience (it inherits the dashboard styles etc)